adoc: added a note on upgrading cri-o

mssola opened the pull request:
cri-o 1.18 requires some manual steps in order to perform the upgrade. This will
be relevant for the 4.3.0 release.

Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

mssola committed:
adoc: added a note on upgrading cri-o

cri-o 1.18 requires some manual steps in order to perform the upgrade. This will
be relevant for the 4.3.0 release.

Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

mjura commented on adoc/MAIN.release-notes.adoc:
```diff
@@ -19,6 +19,20 @@ Release notes usually only list changes that happened between two subsequent rel
+
+=== cri-o
+
+This release upgrades `cri-o` to 1.18, which brings support for Kubernetes 1.18. This new version, though, requires some manual steps for the upgrade, since the configuration files have been migrated from `sysconfig` to `/etc/crio/crio.conf.d/`. In order to upgrade, then, you will have to perform the following commands:
```
from `sysconfig` to `/etc/crio/crio.conf.d/01-caasp.conf`

mjura reviewed (commented):
Only one comment
from `sysconfig` to `/etc/crio/crio.conf.d/01-caasp.conf`

mjura commented on adoc/MAIN.release-notes.adoc:
```diff
@@ -19,6 +19,26 @@ Release notes usually only list changes that happened between two subsequent rel
+
+=== {crio}
+
+This release upgrades `cri-o` to 1.18, which brings support for {kube} 1.18. This new version, though, requires some manual steps for the upgrade, since the configuration files have been migrated from `sysconfig` to `/etc/crio/crio.conf.d/`. In order to upgrade, then, you will have to perform the following commands:
```
```suggestion
This release upgrades `cri-o` to 1.18, which brings support for {kube} 1.18. This new version, though, requires some manual steps for the upgrade, since the configuration files have been migrated from `sysconfig` to `/etc/crio/crio.conf.d/01-caasp.conf`. In order to upgrade, then, you will have to perform the following commands:
```

mssola commented:
@mjura review again :smile:

r0ckarong commented:
We reworked the release notes files to be more manageable. Please create the `release-4.3.0` file and put the content in there. Thanks.

mjura reviewed (approved):
LGTM

evrardjp commented on adoc/release=4.3.0:
```diff
@@ -0,0 +1,13 @@
+rm <CURRENT-CLUSTER-NAME>/addons/cri/default_flags
+----
+
+After this you will be able to perform all the `skuba` commands that you would call normally when upgrading SUSE CaaS Platform.
```
nit: SUSE CaaS Platform should be using a macro.

evrardjp reviewed (approved):
This file will most likely conflict with the other release note for 4.3.0, the hyperkube removal.

I am wondering if these shouldn't be patches on top of each other on a single PR.

Yet, I think it's good.

r0ckarong commented:
> This file will most likely conflict with the other release note for 4.3.0, the hyperkube removal.
> 
> I am wondering if these shouldn't be patches on top of each other on a single PR.
> 
> Yet, I think it's good.

We will have to manually merge all the changes anyway. I don't currently see a better way to do this, it's the same problem as with any code multiple people are working on. We could create a version branch early (right after creating the tag for the "old" one) and add the new file there but that won't solve the problem of multiple patches trying to modify the same file in different ways, same merge effort in the end.

evrardjp commented:
Do not merge until 4.3.0

mssola committed:
Merge remote-tracking branch 'upstream/master'

mssola committed:
Added instructions for a new skuba command

In order to update cri-o and its configuration files, we have introduced a new
subcommand: `skuba cluster upgrade localconfig`. This subcommand is to be called
before upgrading the nodes.

Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

mssola commented:
@r0ckarong I've updated this PR with the subcommand that we have introduced in order to handle the crio update.

mssola commented:
Fixed merge conflicts given base change

Commit:
Add required actions section and wording to k8s 1.18 changes

r0ckarong commented:
Added the "required actions" section back and put a link to the crio update changes. Slight wording review.
